utils/bitfield/src/lib.rs

```diff
@@ -315,3 +315,34 @@ macro_rules! bitfield {
         bitfield!(@iter $($val),*).collect::<$crate::BitField>()
     };
 }
+
+#[cfg(feature = "json")]
+pub mod json {
+    use super::*;
+    use serde::{Deserialize, Deserializer, Serializer};
+
+    pub fn serialize<S>(m: &BitField, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut vec = Vec::new();
+        if !m.is_empty() {
+            m.ranges().for_each(|s| {
+                vec.push(s.start as u8);
+                vec.push(s.end as u8);
+            });
+        } else {
+            vec.push(0);
+        };
+        serializer.serialize_bytes(vec.as_slice())
+    }
+
+    pub fn deserialize<'de, D>(deserializer: D) -> std::result::Result<BitField, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let bitfield_bytes: Vec<usize> = Deserialize::deserialize(deserializer)?;
+        let ranges = ranges_from_bits(bitfield_bytes.iter().cloned());
+        Ok(BitField::from_ranges(ranges))
```

Comment:
you should write a few tests for these, I'm almost certain this is incorrect, because you're assuming it's the range bounds, when it's expecting all set bits in a vector

Reply:
Will implement the tests